warehouse/warehouse/resources/dataset_spec.yaml

```diff
@@ -1540,7 +1587,7 @@ project:
                     - ens1
                 campaign: BGC-v1
                 science_driver: Biogeochemical Cycle
-                cmip_case: CMIP6.CMIP.E3SM-Project.E3SM-1-1.piControl
+                cmip_case: CMIP6.CMIP.E3SM-Project.E3SM-1-1.piControl-ext85yr
```

Comment:
`piControl-ext85yr` is not one of esgf facet names, not sure how this works.I think the extension was appended to the original piControl run when publishing?

Reply:
@TonyB9000 would you clarify on this point?

Reply:
I'm sure that was a copy/paste error.  (for cmip-case).  The "cmip-case" for both "piControl" and "piControl-ext85yr" could be the same.

Perhaps we can eliminate the entire "case"?  But if we are to publish any of the "ex85yr" sets separately of the original, they need their own dataset_ids, and need to be represented separately in the dataset_spec.

Do I understand your question correctly?

Reply:
So -ext85yr in both cmip_case (CMIP6.CMIP.E3SM-Project.E3SM-1-1.piControl and CMIP6.CMIP.E3SM-Project.E3SM-1-1-ECA.piControl) should be removed? or they are fine as is. If i remembered correctly i think those ext85yr runs were stitched to the original publication, and republished.

Reply:
I have reverted that cmip_case to CMIP6.CMIP.E3SM-Project.E3SM-1-1.piControl

Reply:
No - that was the original "quest", but the myriad difficulties that arose when trying to stitch together runs made in differnt years, on different machines, by different people, led to too many obstacles.  So, we (including Sterling) made the decision to publish ALL "extension" sets as separate datasets.  Hence, they need separate dataset_ids.  Therefore, the (project: CMIP6) portion of the dataset_spec reads, in part:

            E3SM-1-1:
                piControl:
                    start: 1850
                    end: 2014
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso
                **piControl-ext85yr**:
                    start: 2015
                    end: 2100
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso
                historical:
                    start: 1850
                    end: 2014
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso
            E3SM-1-1-ECA:
                piControl:
                    start: 1850
                    end: 2014
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso
                **piControl-ext85yr**:
                    start: 2015
                    end: 2100
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso
                historical:
                    start: 1850
                    end: 2014
                    ens:
                        - r1i1p1f1
                    except:
                        - siu
                        - siv
                        - clcalipso

Reply:
Thanks, and i thinks there is another instance for CMIP6.CMIP.E3SM-Project.E3SM-1-1-ECA.piControl https://github.com/E3SM-Project/esgfpub/blob/00586cb6ffb38afe6998b62727eb0d814d6c4b9b/warehouse/warehouse/resources/dataset_spec.yaml#L2087

Reply:
Now I think I need to switch the cmip_case back, or else the "years" will not align.

Reply:
Note:  Technically, once could publish the raw E3SM datasets separately, but publish the CMIP6 sets as a single "1850-2100" body.  But the DataSM will not support that - at present it does not even support generating "pbo/pso" due to the way it attempts to locate the proper (ocean + atmos) datasets needed.  I am struggling with that right now (for the past 48 hours.)

Reply:
Yes, one for "1_1" (CTC) and the other for "1_1_ECA".  The "yaml" format makes that VERY hard to check.  I think using the "list_e3sm_dsids | cut -f1-3 -d. | sort | uniq | grep ext85yr" is the best way to see it.  That yields:

E3SM.1_1_ECA.piControl-ext85yr
E3SM.1_1.piControl-ext85yr

I wish I had the time to produce a flatter structure that I could both read and write, and convert between.  At present, too many codes depend upon the yaml format to dispense with it.  But editing that is a real pain.  Even with those "list_dsid" apps, you cannot see the "cmip_case" values, so you cannot easily tell if each is properly associated with each dataset.